scalalib/src/ScalaModule.scala

```diff
@@ -101,10 +102,23 @@ trait ScalaModule extends JavaModule { outer =>
   def scalaDocPluginIvyDeps = T { scalacPluginIvyDeps() }
 
   /**
-   * Command-line options to pass to the Scala compiler
+   * Mandatory command-line options to pass to the Scala compiler
+   * that shouldn't be removed by overriding `scalacOptions`
+   */
+  protected def mandatoryScalacOptions = T { Seq.empty[String] }
+
+  /**
+   * Command-line options to pass to the Scala compiler defined by the user
+   * Consumers should use `allScalacOptions` to read them.
    */
   def scalacOptions = T { Seq.empty[String] }
 
+  /**
+   * Aggregation of all the options passed to the Scala compiler
+   * Do not override this Target. Override `scalacOptions` instead
```

Comment:
```suggestion
   * Aggregation of all the options passed to the Scala compiler.
   * In most cases, instead of overriding this Target you want to override `scalacOptions` instead.
```